Update user guide (add/edit/delete module, add-person-to-module) (#191)
AY2223S1-CS2103T-T10-1/tp: docs/UserGuide.md

## 1. Quick start

1. Ensure you have Java `11` or above installed in your computer.

2. Download the latest `plannit.jar` [here](https://github.com/AY2223S1-CS2103T-T10-1/tp/releases/).

3. Copy the file to the folder you want to use as the _home folder_ for Plannit. In other words, 

>    the folder which `plannit.jar` is will be the folder in which Plannit will store its data.

Reviewer: ```suggestion
   the folder which `plannit.jar` is in will be the folder in which Plannit will store its data.
```
Missing word here

Author: Thanks for spotting the error. I will fix it.

### 1.1. Command summary

> | Action                                                      | Format                                                                  | Short Description                                                               |

Reviewer: Thanks for updating this

**:information_source: Notes about the command format:**<br>

> * Words in `UPPER_CASE` are the parameters to be supplied by you.<br>

Reviewer: Thanks for changing the phrasing I think this sounds much better!

the parameter will be taken.<br>
  e.g. if you specify `p/81234123 p/99999999`, only `p/99999999` will be taken.

* Any parameter in commands that do not take in parameters (such as
  [`home`](#252-navigate-to-home), [`list-module`](#215-list-module), [`list-person`](#247-list-person),
  [`help`](#26-help) and [`exit`](#27-exiting-the-program)) will be ignored.<br>
  e.g. if the command specifies `home 123`, it will be interpreted as [`home`](#252-navigate-to-home).
</div>

### 2.1. Modules
In Plannit, you can organise tasks and links by academic
module. You can also add study buddies to a module so that you can keep track of who could
potentially be your study buddy.

#### 2.1.1. Add module
You can add a module into Plannit. To prepare for the modules in the upcoming semester, you can add

> your list of modules, one by one, into Plannit.

Reviewer: ```suggestion
your list of modules into Plannit.
```
felt that the one by one was redundant in bringing the idea across though it does make sense in addressing the limitation of add-module

Author: I was thinking that removing "one-by-one" will give the impression that batch addition of modules is possible, where it isn't.

Reviewer: I see where you are coming from. Well to me both are equally well. Hence, I don't really have a strong preference on either of them. We shall adopt your description!

```
add-module m/CS2104
```
OR
```
add-module m/CS2104 t/
```
In either of the above examples, we are adding a module `CS2104` without a title.

```
add-module m/CS2104 t/Programming Language Concepts
```
In the above example, we are adding a module `CS2104` which has the title `Programming Language Concepts`.
Here's a screenshot of Plannit before and after executing the command:

| Before executing the command                                                                                                                                           | After executing the command                                                                                                                                                                                                            |
|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | {::nomarkdown}<p align="center"><img src="images/add-module-before.png" width="300"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.</p>{:/} | {::nomarkdown}<p align="center"><img src="images/add-module-after.png" width="300"/></p><p>A message will appear indicating that <code>CS2104</code> has been added. Observe that `CS2104` has been added to the modules list.</p>{:/} |

Reviewer 2: ```suggestion
| {::nomarkdown}<p align="center"><img src="images/add-module-before.png" width="300"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.</p>{:/} | {::nomarkdown}<p align="center"><img src="images/add-module-after.png" width="300"/></p><p>A message will appear indicating that <code>CS2104</code> has been added. Observe that <code>CS2104</code> has been added to the modules list.</p>{:/} |
```
Missing tag?

In the above example, we are adding a module `CS2104` which has the title `Programming Language Concepts`.
Here's a screenshot of Plannit before and after executing the command:

{::nomarkdown}
<table>
    <tr>
        <td width="50%">Before executing the command</td>
        <td width="50%">After executing the command</td>
    </tr>
    <tr>

>         <td><p align="center"><img src="images/add-module-before.png" width="300"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.</p></td>

Reviewer 2: ```suggestion
        <td><p align="center"><img src="images/add-module-before.png" width="300"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.<br></p></td>
```
Perhaps you could add a newline character so that the two images will be aligned vertically?
![image](https://user-images.githubusercontent.com/83915879/200135075-15f41547-1529-4daa-bd80-e4041aab28f0.png)

Author: `<td style="vertical-align: top">` did the trick.

delete-module m/CS2104
```
In the above example, we are deleting module `CS2104` from Plannit.
Here's a screenshot of Plannit before and after executing the command:

{::nomarkdown}
<!-- Solution adapted from https://stackoverflow.com/questions/5671687/i-want-to-align-the-text-in-a-td-to-the-top and https://stackoverflow.com/questions/2919980/table-with-100-width-with-equal-size-columns -->
<table>
    <tr>
        <td width="50%"><b>Before executing the command</b></td>
        <td width="50%"><b>After executing the command</b></td>
    </tr>
    <tr>
        <td style="vertical-align: top"><p align="center"><img src="images/add-module-before.png"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.</p></td>
        <td style="vertical-align: top"><p align="center"><img src="images/delete-module-after.png"/></p><p>A message will appear indicating that <code>CS2104</code> has been deleted. Observe that <code>CS2104</code> has been removed from the module list.</p></td>
    </tr>
</table>
{:/}

<div markdown="span" class="alert alert-info">:information_source: **Note:**<br>
Deleting a module will bring you back to the home page.
</div>

#### 2.1.3. Edit module

> You can edit a module on Plannit using the `edit-module` command. This is useful when you want 
> to add a module title when it is available, or make corrections due to an accidental typo in one 
> of the modules, without having to delete the entire module with links and tasks, and recreating it.

Reviewer: ```suggestion
You can edit a module on Plannit using the `edit-module` command. In this way, you can modify module titles and descriptions without having to delete and recreate the entire module, together with its associated links, tasks and contacts.
```
Just a suggestion! thought it sounds better like this

Author: Yes, I agree. I will fix this.

|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | {::nomarkdown}<p align="center"><img src="images/edit-module-before.png" width="300"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.</p>{:/} | {::nomarkdown}<p align="center"><img src="images/edit-module-after.png" width="300"/></p><p>A message will appear indicating that a module has been edited. Observe that the third module in the modules list has its module code changed to <code>CS2104</code> and its module title changed to `Programming Language Concepts`. Note that the module index has changed since Plannit automatically sorts modules in ascending order of module code.</p>{:/} |

Reviewer 2: ```suggestion
| {::nomarkdown}<p align="center"><img src="images/edit-module-before.png" width="300"/></p><p>Enter the command into the command box and hit <code>ENTER</code>.</p>{:/} | {::nomarkdown}<p align="center"><img src="images/edit-module-after.png" width="300"/></p><p>A message will appear indicating that a module has been edited. Observe that the third module in the modules list has its module code changed to <code>CS2104</code> and its module title changed to <code>Programming Language Concepts</code>. Note that the module index has changed since Plannit automatically sorts modules in ascending order of module code.</p>{:/} |
```
Missing tag?

Author: Yes, that is right, thank you.

You will leave the home page after executing the `goto` command. This is different from the
behavior of [`find-module`](#214-find-module) command. <br> <br>
After using the `goto` command, usage of the following commands will be restricted:<br>

> 

Reviewer: Thanks for helping me with my part to ensure nicer formatting of the user guide!
